### PR TITLE
Add java ver and hover to versions subcommand

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
Add the Java version to `/sf versions` and add some hover text to it if using a version lower than 11. I also added hover text to disabled plugins.

Image of full versions:
image

Image of Java hover text (Need suggestions on the text):
image

Image of disabled plugin hover text:
image

## Changes
* Added "Java" to `/sf versions`
* Redone the `/sf versions` message to be a TextComponent
* Added hover text for the Java field
* Added hover text for any disabled plugins

## Related Issues
N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java [hunk 1]
```diff
@@ -28,35 +32,75 @@ public void onExecute(@Nonnull CommandSender sender, @Nonnull String[] args) {
...
             SlimefunPlugin.getLocalization().sendMessage(sender, "messages.no-permission", true);
         }
     }
 
+    private void addJavaVersion(@Nonnull ComponentBuilder builder) {
+        String javaVer = System.getProperty("java.version");
+        if (javaVer.startsWith("1.")) {
+            javaVer = javaVer.substring(2);
+        }
+        if (javaVer.indexOf('.') != -1) { // If it's like 11.0.1.3 or 8.0_275
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:67]: No comment and code on the same line!

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java [hunk 2]
```diff
@@ -28,35 +32,75 @@ public void onExecute(@Nonnull CommandSender sender, @Nonnull String[] args) {
...
+
+        for (Plugin plugin : addons) {
+            String version = plugin.getDescription().getVersion();
+
+            if (Bukkit.getPluginManager().isPluginEnabled(plugin)) {
+                builder.append("\n  " + plugin.getName()).color(ChatColor.GREEN)
+                    .append(" v" + version).color(ChatColor.DARK_GREEN);
+            } else {
+                HoverEvent hover = new HoverEvent(HoverEvent.Action.SHOW_TEXT,
+                    new Text("Plugin is disabled. Check the console for an error and report on their issue tracker."));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:97]: Should probably do an `instanceof SlimefunAddon` check here and link the issues tracker then.

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java [hunk 3]
```diff
@@ -28,35 +32,75 @@ public void onExecute(@Nonnull CommandSender sender, @Nonnull String[] args) {
...
-                String version = plugin.getDescription().getVersion();
+            addPluginVersions(builder);
 
-                if (Bukkit.getPluginManager().isPluginEnabled(plugin)) {
-                    sender.sendMessage(ChatColor.GREEN + " " + plugin.getName() + ChatColor.DARK_GREEN + " v" + version);
-                } else {
-                    sender.sendMessage(ChatColor.RED + " " + plugin.getName() + ChatColor.DARK_RED + " v" + version);
-                }
-            }
+            System.out.println(ComponentSerializer.toString(builder.create()));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:55]: Sysout :NotLikeThis:

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java [hunk 4]
```diff
@@ -28,35 +32,75 @@ public void onExecute(@Nonnull CommandSender sender, @Nonnull String[] args) {
...
-                sender.sendMessage(ChatColors.color("&aMetrics build: &2#" + SlimefunPlugin.getMetricsService().getVersion()));
+                builder.append("Metrics build: ").color(ChatColor.GREEN)
+                    .append("#" + SlimefunPlugin.getMetricsService().getVersion() + '\n').color(ChatColor.DARK_GREEN);
             }
 
+            addJavaVersion(builder);
+
             if (SlimefunPlugin.getRegistry().isBackwardsCompatible()) {
-                sender.sendMessage(ChatColor.RED + "Backwards compatibility enabled!");
+                builder.append("Backwards compatibility enabled!\n").color(ChatColor.RED);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:48]: Should probably also add a tooltip here.

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java [hunk 5]
```diff
@@ -28,35 +32,75 @@ public void onExecute(@Nonnull CommandSender sender, @Nonnull String[] args) {
...
+        Collection<Plugin> addons = SlimefunPlugin.getInstalledAddons();
+        builder.append("Installed Addons: ").color(ChatColor.GRAY)
+            .append("(" + addons.size() + ")").color(ChatColor.DARK_GRAY);
+
+        for (Plugin plugin : addons) {
+            String version = plugin.getDescription().getVersion();
+
+            if (Bukkit.getPluginManager().isPluginEnabled(plugin)) {
+                builder.append("\n  " + plugin.getName()).color(ChatColor.GREEN)
+                    .append(" v" + version).color(ChatColor.DARK_GREEN);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:94]: We could add a hover event with the author and bug tracker too, not really necessary though, just an idea.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
All of this formatting :NotLikeThis:

**Comment by TheBusyBiscuit** [comment]
whoops, did I accidentally assign you instead of me?
*glares...*

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java (hunk 4 above)
Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:48]: yeah i was going to but really wasn't sure how to write it.

"If you don't have items older than Summer 2019 you should disable this to increase performance"
maybe?

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:48]: "Backwards compatibility has a negative impact on performance!
We recommend you to disable this setting unless your Server still
has legacy Slimefun items (from before summer 2019) in circulation."

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java [hunk 6]
```diff
@@ -3,13 +3,20 @@
...
 import javax.annotation.Nonnull;
+
+import io.github.thebusybiscuit.slimefun4.api.SlimefunAddon;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
+import net.md_5.bungee.api.ChatColor;
+import net.md_5.bungee.api.chat.ClickEvent;
+import net.md_5.bungee.api.chat.ComponentBuilder;
+import net.md_5.bungee.api.chat.HoverEvent;
+import net.md_5.bungee.api.chat.hover.content.Text;
+import net.md_5.bungee.chat.ComponentSerializer;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:14]: Unused imports

```suggestion
import net.md_5.bungee.api.ChatColor;
import net.md_5.bungee.api.chat.ClickEvent;
import net.md_5.bungee.api.chat.ComponentBuilder;
import net.md_5.bungee.api.chat.HoverEvent;
import net.md_5.bungee.api.chat.hover.content.Text;
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java [hunk 7]
```diff
@@ -28,35 +35,107 @@ public void onExecute(@Nonnull CommandSender sender, @Nonnull String[] args) {
...
+            String version = plugin.getDescription().getVersion();
 
-            for (Plugin plugin : addons) {
-                String version = plugin.getDescription().getVersion();
+            if (Bukkit.getPluginManager().isPluginEnabled(plugin)) {
+                HoverEvent hoverEvent;
+                ClickEvent clickEvent = null;
+                if (plugin instanceof SlimefunAddon) {
+                    hoverEvent = new HoverEvent(HoverEvent.Action.SHOW_TEXT, new Text(
+                        "Author: " + String.join(", ", plugin.getDescription().getAuthors())
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:107]:
```suggestion
                        "Author(s): " + String.join(", ", plugin.getDescription().getAuthors())
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java [hunk 8]
```diff
@@ -28,35 +35,107 @@ public void onExecute(@Nonnull CommandSender sender, @Nonnull String[] args) {
...
+                        + "\nClick to open the Bug Tracker"
+                    ));
+                    clickEvent = new ClickEvent(ClickEvent.Action.OPEN_URL, ((SlimefunAddon) plugin).getBugTrackerURL());
+                } else {
+                    hoverEvent = new HoverEvent(HoverEvent.Action.SHOW_TEXT, new Text(
+                        "Author: " + String.join(", ", plugin.getDescription().getAuthors())
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/VersionsCommand.java:113]:
```suggestion
                        "Author(s): " + String.join(", ", plugin.getDescription().getAuthors())
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]
Can always do some styling changes, like color, in post.
But looks good to me.

**Review by Sfiguz7 — APPROVED** [review]
LGTM (Addon author info which everyone will miss and still report in #bug-discussions :D)

**Review by Sfiguz7 — APPROVED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**